aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -325,108 +343,128 @@ def export_tree(
 
-        foreign_fields = {
-            k: v
-            for k, v in all_fields_info[entity_name].items()
-            # all_fields_info[model_name].items()
-            if 'requires' in v
-        }
+        if not silent:
+            progress_bar.set_description_str('Exporting {}s'.format(entity_name), refresh=False)
+            progress_bar.update()
+
+        foreign_fields = {k: v for k, v in all_fields_info[entity_name].items() if 'requires' in v}
 
         for value in foreign_fields.values():
             ref_model_name = value['requires']
             fill_in_query(partial_query, entity_name, ref_model_name, [entity_name], entity_separator)
 
         for temp_d in partial_query.iterdict():
-            for k in temp_d.keys():
+            for key in temp_d:
                 # Get current entity
-                current_entity = k.split(entity_separator)[-1]
+                current_entity = key.split(entity_separator)[-1]
 
                 # This is a empty result of an outer join.
                 # It should not be taken into account.
-                if temp_d[k]['id'] is None:
+                if temp_d[key]['id'] is None:
                     continue
 
-                temp_d2 = {
-                    temp_d[k]['id']:
+                export_data[current_entity].update({
+                    temp_d[key]['id']:
                     serialize_dict(
-                        temp_d[k], remove_fields=['id'], rename_fields=model_fields_to_file_fields[current_entity]
+                        temp_d[key], remove_fields=['id'], rename_fields=model_fields_to_file_fields[current_entity]
                     )
-                }
-                try:
-                    export_data[current_entity].update(temp_d2)
-                except KeyError:
-                    export_data[current_entity] = temp_d2
+                })
+
+    if not silent:
+        # Close progress up until this point in order to print properly
+        progress_bar.leave = False
+        progress_bar.close()
 
     #######################################
     # Manually manage attributes and extras
     #######################################
-    # I use .get because there may be no nodes to export
-    all_nodes_pk = list()
-    if NODE_ENTITY_NAME in export_data:
-        all_nodes_pk.extend(export_data.get(NODE_ENTITY_NAME).keys())
+    # Pointer. Renaming, since Nodes have now technically been retrieved and "stored"
+    all_node_pks = node_ids_to_be_exported
 
-    if sum(len(model_data) for model_data in export_data.values()) == 0:
-        if not silent:
-            print('No nodes to store, exiting...')
+    model_data = sum(len(model_data) for model_data in export_data.values())
+    if not model_data:
+        if not silent or debug:
+            print('Nothing to store, exiting...')
         return
 
+    if not silent or debug:
+        print('Exporting a total of {} database entries, of which {} are Nodes.'.format(model_data, len(all_node_pks)))
```

Comment:
```suggestion
        print('Exporting {} database entries ({} Nodes).'.format(model_data, len(all_node_pks)))
```
(otherwise doesn't fit left of the bar)

Reply:
This is not supposed to be in the progress bar. This is a permanent print.

Check the export/import in normal mode.

Reply:
I see, so I guess what is wrong here is the if statement: this is printed in non-debug mode!

Reply:
It's supposed to be present if we're _not_ in silent mode (default) OR in debug mode.